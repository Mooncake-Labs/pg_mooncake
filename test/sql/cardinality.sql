CREATE TABLE t (a int) USING columnstore;
INSERT INTO t SELECT * FROM generate_series(1, 100);
EXPLAIN SELECT * FROM t;
INSERT INTO t SELECT * FROM generate_series(1, 200);
EXPLAIN SELECT * FROM t;
DROP TABLE t;
