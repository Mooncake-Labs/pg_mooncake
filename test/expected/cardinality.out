CREATE TABLE t (a int) USING columnstore;
INSERT INTO t SELECT * FROM generate_series(1, 100);
EXPLAIN SELECT * FROM t;
                             QUERY PLAN                             
--------------------------------------------------------------------
 Custom Scan (MooncakeDuckDBScan)  (cost=0.00..0.00 rows=0 width=0)
   DuckDB Execution Plan: 
 
 ┌───────────────────────────┐
 │     COLUMNSTORE_SCAN      │
 │    ────────────────────   │
 │         Function:         │
 │      COLUMNSTORE_SCAN     │
 │                           │
 │       Projections: a      │
 │                           │
 │         ~100 Rows         │
 └───────────────────────────┘
 
 
(15 rows)

INSERT INTO t SELECT * FROM generate_series(1, 200);
EXPLAIN SELECT * FROM t;
                             QUERY PLAN                             
--------------------------------------------------------------------
 Custom Scan (MooncakeDuckDBScan)  (cost=0.00..0.00 rows=0 width=0)
   DuckDB Execution Plan: 
 
 ┌───────────────────────────┐
 │     COLUMNSTORE_SCAN      │
 │    ────────────────────   │
 │         Function:         │
 │      COLUMNSTORE_SCAN     │
 │                           │
 │       Projections: a      │
 │                           │
 │         ~300 Rows         │
 └───────────────────────────┘
 
 
(15 rows)

DROP TABLE t;
